src/when.js

```diff
@@ -0,0 +1,34 @@
+var _curry2 = require('./internal/_curry2');
+var _identity = require('./internal/_identity');
+var ifElse = require('./ifElse');
+
+
+/**
+ * Creates a function that will first pass its argument to the given predicate
+ * function. If the predicate is satisfied, the function will return the result
+ * of calling the `whenTrue` function with the same argument. If the predicate
+ * is not satisfied, the argument is returned as is.
+ *
+ * @func
+ * @memberOf R
+ * @category Logic
+ * @see R.ifElse, R.unless
+ * @sig (a -> Boolean) -> (a -> a) -> (a -> a)
+ * @param {Function} condition A predicate function
+ * @param {Function} whenTrue  A function to invoke when the `condition`
+ *                             evaluates to a truthy value.
+ * @return {Function} A new function that will process the `whenTrue` function
+ *                    depending upon the result of the `condition` predicate.
+ * @example
+ *
+ *      // truncate :: String -> String
+ *      var truncate = R.when(
+ *        R.propSatisfies(R.gt(R.__, 10), 'length'),
+ *        R.pipe(R.take(10), R.append('…'), R.join(''))
+ *      );
+ *      truncate('12345');         //=> '12345'
+ *      truncate('0123456789ABC'); //=> '0123456789…'
+ */
+module.exports = _curry2(function when(pred, whenTrueFn) {
+  return ifElse(pred, whenTrueFn, _identity);
```

Comment:
I'm wondering whether it's a bit unnecessary to use `ifElse` for the implementation here. These are unary functions by design, so could be implemented as simply as:
```js
_curry3(function when(pred, whenTrueFn, x) {
  return pred(x) ? whenTrueFn(x) : x;
});
```